### PR TITLE
fix(dedicated): replace billing payments apiv7 calls with iceberg

### PR DESCRIPTION
**Pull request by cbourgois**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/kill-apiv7`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix MANAGER-5631
| License          | BSD 3-Clause

## Description

Replace billing payments calls (list & details) apiv7 calls with iceberg

I have also moved from `function` to `class`(with `@ngInject`) but don't refactor the modules/component lifecycles.

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/billing/main/payments/billing-payments.service.js [hunk 1]
```diff
@@ -1,158 +1,72 @@
...
+      return this.getOperationsDetails(id, bill.billId, bill.orderId).then(
+        (operations) => {
+          if (!operations || operations.length === 0) {
+            throw new Error('No operation for a bill concerned by a deposit');
+          }
+          if (operations.length > 1) {
+            throw new Error(
+              'More than one operation for a bill concerned by a deposit',
+            );
+          }
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/billing/main/payments/billing-payments.service.js:43]: Do we catch or do something with those error ?

Comment (cbourgois) [reply packages/manager/apps/dedicated/client/app/billing/main/payments/billing-payments.service.js:43]: Will check. Not in the scope of this PR. But if I can do something, I'll do it ;)

**Review by Jisay — DISMISSED** [review]

**Review by antleblanc — DISMISSED** [review]

**Comment by cbourgois** [comment]
Opened in Septembre 2020, rebased now.

**Review by antleblanc — APPROVED** [review]

**Review by y4nnL — APPROVED** [review]

File: packages/manager/modules/billing/src/main/payments/billing-payments.controller.js [hunk 2]
```diff
@@ -1,45 +1,75 @@
...
-  atInternet,
-  coreConfig,
-  OvhApiMe,
-) {
-  this.loadPayments = ($config) => {
-    let request = OvhApiMe.Deposit()
-      .v7()
+  $onInit() {
+    this.payments = [];
+    if (this.coreConfig.getRegion() === 'US') {
```
Comment (marie-j) [inline packages/manager/modules/billing/src/main/payments/billing-payments.controller.js:25]: Cannot be handled by feature flipping ?

**Review by marie-j — APPROVED** [review]

File: packages/manager/modules/billing/src/main/payments/billing-payments.controller.js (hunk 2 above)
Comment (cbourgois) [reply packages/manager/modules/billing/src/main/payments/billing-payments.controller.js:25]: Good idea, will be done in another PR.

**Pull request merged**